roles/rhods_test_jupyterlab/tasks/main.yml

```diff
@@ -174,9 +174,18 @@
   environment:
     KUBECONFIG: '{{ sut_cluster_kubeconfig }}'
   command: |
-    oc delete pod --all -n  {{ rhods_notebook_namespace }}
-    oc delete pvc --all -n {{ rhods_notebook_namespace }}
-    oc delete ev --all -n {{ rhods_notebook_namespace }}
+    oc get pods -o wide -n {{ rhods_notebook_namespace }} | \
+      grep -i -E {{ rhods_test_jupyterlab_username_prefix }} | \
+      awk '{print $1}' | \
+      xargs oc delete pod
+    oc get pvc -o wide -n {{ rhods_notebook_namespace }} | \
+      grep -i -E {{ rhods_test_jupyterlab_username_prefix }} | \
+      awk '{print $1}' | \
+      xargs oc delete pvc
+    oc get ev -o wide -n {{ rhods_notebook_namespace }} | \
+      grep -i -E {{ rhods_test_jupyterlab_username_prefix }} | \
+      awk '{print $1}' | \
+      xargs oc delete ev
```

Comment:
I think we should delete all the Events before launching the test, so that we can capture all the events generated during the test. 
Same for Pods, we should delete them all to be sure that the cluster is clean when we start.
PVCs are diffferent because 1/ they don't actively consume resources (only storage) and 2/ they may store an important state

Reply:
I am not sure I would delete active notebook pods belonging to users. @erwangranger would you please chime in?

Reply:
I think the answer is different depending on the situation. 

if it's after the test, we want to make sure the cleanup is thorough. A running pod isn't the end of the world, as it will eventually stop. But a stray PVC will cost money for a long time. So, for the post-test, I would first kill off all the pods, and then delete all the PVCs. (If you do both at once, I'm not sure that you'd be able to delete a PVC that is potentially still mounted onto a pod). 

For the pre-test cleanup, I think it matters less. The tests should be able to cope with a pre-existing PVC, or even a pre-running pod. 

In both cases, however, do we have any assurance that `rhods_test_jupyterlab_username_prefix` will be unique enough that it will only match things that were created by our code?

Reply:
It's currently set to "testuser". So in theory it shouldn't match any living user, but I can foresee some situations (PoCs) where this cleanup could do the wrong thing.

Reply:
it's easy to change this value, see there:
https://github.com/openshift-psap/ci-artifacts/blob/master/testing/ods/common.sh#L28

so we can easily change it to something like `psap_testuser` if we learn (before running the test) that `testuser` might conflict with things already in use

Reply:
Makes sense. https://github.com/openshift-psap/ci-artifacts/pull/426